bertopic/_bertopic.py

```diff
@@ -3565,168 +3593,62 @@ def _zeroshot_topic_modeling(self, documents: pd.DataFrame, embeddings: np.ndarr
         assigned_documents["ID"] = range(len(assigned_documents))
         assigned_embeddings = embeddings[assigned_ids]
 
+        # Check that if a number of topics was specified, it exceeds the number of zeroshot topics matched
+        num_zeroshot_topics = len(assigned_documents["Topic"].unique())
+        if self.nr_topics and not self.nr_topics > num_zeroshot_topics:
+            raise ValueError(f'The set nr_topics ({self.nr_topics}) must exceed the number of matched zero-shot topics '
+                             f'({num_zeroshot_topics}). Consider raising nr_topics or raising the '
+                             f'zeroshot_min_similarity ({self.zeroshot_min_similarity}).')
+
         # Select non-assigned topics to be clustered
         documents = documents.iloc[non_assigned_ids]
         documents["Old_ID"] = documents["ID"].copy()
         documents["ID"] = range(len(documents))
         embeddings = embeddings[non_assigned_ids]
 
-        # If only matches were found
-        if len(non_assigned_ids) == 0:
-            return None, None, assigned_documents, assigned_embeddings
         logger.info("Zeroshot Step 1 - Completed \u2713")
         return documents, embeddings, assigned_documents, assigned_embeddings
 
     def _is_zeroshot(self):
         """ Check whether zero-shot topic modeling is possible
 
-        * There should be a cluster model used
         * Embedding model is necessary to convert zero-shot topics to embeddings
         * Zero-shot topics should be defined
         """
-        if self.zeroshot_topic_list is not None and self.embedding_model is not None and type(self.hdbscan_model) != BaseCluster:
+        if self.zeroshot_topic_list is not None and self.embedding_model is not None:
             return True
         return False
 
     def _combine_zeroshot_topics(self,
                                  documents: pd.DataFrame,
+                                 embeddings: np.ndarray,
                                  assigned_documents: pd.DataFrame,
-                                 embeddings: np.ndarray) -> Union[Tuple[np.ndarray, np.ndarray], np.ndarray]:
-        """ Combine the zero-shot topics with the clustered topics
-
-        There are three cases considered:
-        * Only zero-shot topics were found which will only return the zero-shot topic model
-        * Only clustered topics were found which will only return the clustered topic model
-        * Both zero-shot and clustered topics were found which will return a merged model
-          * This merged model is created using the `merge_models` function which will ignore
-            the underlying UMAP and HDBSCAN models
-
-        Arguments:
-            documents: Dataframe with documents and their corresponding IDs
-            assigned_documents: Dataframe with documents and their corresponding IDs
-                                that were assigned to a zero-shot topic
-            embeddings: The document embeddings
```

Comment:
I would like to keep the docstrings here as it helps me (and others) develop. We could also move this to `._zeroshot_topic_modeling` and add some small docstrings here to indicate that this is only run when we have both zeroshot topics and non-zeroshot topics.

Reply:
I had removed the docstring when making big changes and forgotten to write a new one. Added.